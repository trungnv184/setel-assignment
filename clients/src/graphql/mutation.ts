import { gql } from '@apollo/client';

const ADD_PRODUCT_TO_CART = gql`
  mutation AddToCart($cartInput: CartInput) {
    addToCart(cartInput: $cartInput) {
      id
      name
      url
      price
    }
  }
`;

const CREATE_ORDER_PRODUCTS = gql`
  mutation CreateOrder($orderInput: OrderInput) {
    createOrder(orderInput: $orderInput) {
      orderItems {
        id
        name
        url
        price
        quantity
        createdDate
      }
      metadata {
        firstName
        lastName
        address
        phoneNumber
      }
      notes
    }
  }
`;

export { ADD_PRODUCT_TO_CART, CREATE_ORDER_PRODUCTS };
