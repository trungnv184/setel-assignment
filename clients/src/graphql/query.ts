import { gql } from '@apollo/client';

const GET_PRODUCTS = gql`
  query GetProducts {
    products {
      id
      name
      price
      url
      rating
    }
  }
`;

const GET_PRODUCTS_IN_CART = gql`
  query GetProductsCart {
    getProductsCart {
      id
      name
      price
      url
      quantity
    }
  }
`;

const GET_ORDER_BY_ID = gql`
  query GetOrder($orderId: String) {
    getOrder(orderId: $orderId) {
      id
      orderItems {
        id
        name
        price
        url
        quantity
      }
      metadata {
        firstName
        lastName
        address
        phoneNumber
      }
      notes
      state
      createdDate
      updatedDate
    }
  }
`;

const GET_ORDER_STATUS = gql`
  query GetOrderStatus($orderId: String) {
    getOrder(orderId: $orderId) {
      state
    }
  }
`;

export { GET_PRODUCTS, GET_PRODUCTS_IN_CART, GET_ORDER_BY_ID, GET_ORDER_STATUS };
