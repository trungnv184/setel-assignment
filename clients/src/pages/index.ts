export { default as Orders } from './Orders/Orders';
export { default as Products } from './Products/Products';
export { default as Carts } from './Carts/Carts';
export { default as OrderDetail } from './OrderDetail/OrderDetail';
