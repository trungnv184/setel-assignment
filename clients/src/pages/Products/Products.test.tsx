import { render, screen, waitFor } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { PRODUCTS_MOCK } from 'test/mocks/products.mock';
import { Products } from 'pages';
import { GET_PRODUCTS } from 'graphql/query';

let isCalled = false;
const mocks = [
  {
    request: {
      query: GET_PRODUCTS,
    },
    result: () => {
      isCalled = true;
      return {
        data: {
          products: PRODUCTS_MOCK,
        },
      };
    },
  },
];
describe('ProductsPage', () => {
  test('Should render products page', async () => {
    const { container } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Products />
      </MockedProvider>
    );

    await waitFor(() => {
      expect(isCalled).toBeTruthy();
    });

    const bookNameElements = screen.getAllByTestId('book-name');
    expect(bookNameElements.length).toBe(2);
    expect(bookNameElements[0].textContent).toBe('Kiên trì hay từ bỏ');
    expect(bookNameElements[1].textContent).toBe('Rèn luyện tư duy phản biện');

    expect(container).toMatchSnapshot();
  });
});
