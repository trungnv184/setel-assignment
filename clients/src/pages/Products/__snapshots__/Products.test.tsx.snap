// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsPage Should render products page 1`] = `
<div>
  <div
    class="ui six cards"
  >
    <div
      class="ui raised card"
    >
      <div
        class="large image"
      >
        <img
          src="https://salt.tikicdn.com/cache/280x280/ts/product/6e/41/a2/721878ef916620dbcc7ff569adcd577d.jpg"
        />
      </div>
      <div
        class="content"
      >
        <div
          class="header"
          data-testid="book-name"
        >
          Kiên trì hay từ bỏ
        </div>
        <div
          class="meta"
        >
          <strong>
            200.000 ₫
          </strong>
        </div>
      </div>
      <div
        class="extra content"
      >
        <div
          class="ui star rating"
          role="radiogroup"
          tabindex="-1"
        >
          <i
            aria-checked="false"
            aria-posinset="1"
            aria-setsize="5"
            class="active icon"
            role="radio"
            tabindex="0"
          />
          <i
            aria-checked="false"
            aria-posinset="2"
            aria-setsize="5"
            class="active icon"
            role="radio"
            tabindex="0"
          />
          <i
            aria-checked="false"
            aria-posinset="3"
            aria-setsize="5"
            class="active icon"
            role="radio"
            tabindex="0"
          />
          <i
            aria-checked="false"
            aria-posinset="4"
            aria-setsize="5"
            class="active icon"
            role="radio"
            tabindex="0"
          />
          <i
            aria-checked="true"
            aria-posinset="5"
            aria-setsize="5"
            class="active icon"
            role="radio"
            tabindex="0"
          />
        </div>
        <button
          class="ui red right floated button"
        >
          Buy
        </button>
      </div>
    </div>
    <div
      class="ui raised card"
    >
      <div
        class="large image"
      >
        <img
          src="https://salt.tikicdn.com/cache/w444/ts/product/22/cb/a9/524a27dcd45e8a13ae6eecb3dfacba7c.jpg"
        />
      </div>
      <div
        class="content"
      >
        <div
          class="header"
          data-testid="book-name"
        >
          Rèn luyện tư duy phản biện
        </div>
        <div
          class="meta"
        >
          <strong>
            200.000 ₫
          </strong>
        </div>
      </div>
      <div
        class="extra content"
      >
        <div
          class="ui star rating"
          role="radiogroup"
          tabindex="-1"
        >
          <i
            aria-checked="false"
            aria-posinset="1"
            aria-setsize="5"
            class="active icon"
            role="radio"
            tabindex="0"
          />
          <i
            aria-checked="false"
            aria-posinset="2"
            aria-setsize="5"
            class="active icon"
            role="radio"
            tabindex="0"
          />
          <i
            aria-checked="true"
            aria-posinset="3"
            aria-setsize="5"
            class="active icon"
            role="radio"
            tabindex="0"
          />
          <i
            aria-checked="false"
            aria-posinset="4"
            aria-setsize="5"
            class="icon"
            role="radio"
            tabindex="0"
          />
          <i
            aria-checked="false"
            aria-posinset="5"
            aria-setsize="5"
            class="icon"
            role="radio"
            tabindex="0"
          />
        </div>
        <button
          class="ui red right floated button"
        >
          Buy
        </button>
      </div>
    </div>
  </div>
</div>
`;
